### PR TITLE
Wire extension updates in quarkus update

### PR DESCRIPTION
**Pull request by gsmet**
This is a relatively naive and simple approach but it seems to work OK.

Note that this is a preliminar effort and it is not handling all the
possibilities.
Handling the other possibilities might be problematic given how we apply
the recipes to a root module of a multi-module project.

@aloubyansky I would be interested in your feedback.

/cc @ia3andy @aloubyansky

### CODE REVIEW

**Comment by gsmet** [comment]
@aloubyansky could you have a look at this one? I think we will want it in 3.2 maybe.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdates.java [hunk 1]
```diff
@@ -48,6 +52,24 @@ public static FetchResult createRecipe(MessageWriter log, Path target, MavenArti
                 break;
         }
 
+        for (List<ExtensionUpdateInfo> nonPlatformExtensionsUpdates : request.projectExtensionsUpdateInfo
+                .getNonPlatformExtensions().values()) {
+            for (ExtensionUpdateInfo nonPlatformExtensionsUpdate : nonPlatformExtensionsUpdates) {
+                if (nonPlatformExtensionsUpdate.getCurrentDep().isPlatformExtension()) {
+                    // add, my understanding is that we should define the version? As a dependency, as a managed one?
+                    // not completely sure how to make it work for a multi-module project?
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdates.java:60]: Should we log something in this case? AFAIU, we are not applying any update in this case.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdates.java [hunk 2]
```diff
@@ -48,6 +52,24 @@ public static FetchResult createRecipe(MessageWriter log, Path target, MavenArti
...
+                } else if (nonPlatformExtensionsUpdate.getRecommendedDependency().isPlatformExtension()) {
+                    // remove, decide what to do here, should we remove the version given it is now managed? Will OpenRewrite support that?
+                    // not completely sure how to make it work for a multi-module project?
```
Comment (aloubyansky) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdates.java:63]: Same here?

**Comment by gsmet** [comment]
@aloubyansky I think we should get this one in even if not perfect. I'm not sure I fully understand the two other cases so if you want something logged you would have to finish this.
Also, you are logging these at the very beginning of the process.

**Comment by aloubyansky** [comment]
Ok, those are not usual cases.

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**